Src/library/net/network/socket/network_socket.e

```diff
@@ -194,47 +210,82 @@ feature -- Status setting
 			socket_exists: exists
 		do
 			c_set_sock_opt_int (descriptor, level_sol_socket, so_reuse_addr, 1)
-		end;
+		end
 
 	do_not_reuse_address
 			-- Turn `reuse_address' option off.
 		require
 			socket_exists: exists
 		do
 			c_set_sock_opt_int (descriptor, level_sol_socket, so_reuse_addr, 0)
-		end;
+		end
 
 	set_timeout (n: INTEGER)
 			-- Set timeout to `n' seconds.
+		obsolete
+			"Use `set_timeout_ns (ns)` [2018-01-15]"
 		require
 			non_negative: n >= 0
 		do
-			if n > 0 then
-				timeout := n
-			else
-				timeout := default_timeout
-			end
+			set_timeout_ns (n.to_natural_64)
 		ensure
 			timeout_set: timeout = n or timeout = default_timeout
 		end
 
+	set_timeout_ns (a_timeout_nanoseconds: NATURAL_64)
+			-- Set timeout with to `a_timeout_nanoseconds` nanoseconds.
+			-- Warning: the timeout granularity of the platform may not be nanoseconds, but micro or milliseconds.
+		require
+			is_valid_timeout_ns: is_valid_timeout_ns (a_timeout_nanoseconds)
+		do
+			if a_timeout_nanoseconds = 0 then
```

Comment:
I believe this case is ruled out by the precondition.

Reply:
actually not since it is ok to have 0 per the precondition